mesonbuild/rewriter.py

```diff
@@ -562,17 +598,17 @@ def process_kwargs(self, cmd):
 
             # Remove the key from the kwargs
             if cmd['operation'] == 'delete':
-                if key in arg_node.kwargs:
+                if key in kwargs:
                     mlog.log('  -- Deleting', mlog.bold(key), 'from the kwargs')
-                    del arg_node.kwargs[key]
+                    del kwargs[key]
                     num_changed += 1
                 else:
                     mlog.log('  -- Key', mlog.bold(key), 'is already deleted')
                 continue
 
-            if key not in arg_node.kwargs:
-                arg_node.kwargs[key] = None
-            modifier = kwargs_def[key](arg_node.kwargs[key])
+            if key not in kwargs:
+                kwargs[key] = None
+            modifier = kwargs_def[key](kwargs[key])
             if not modifier.can_modify():
                 mlog.log('  -- Skipping', mlog.bold(key), 'because it is to complex to modify')
```

Comment:
```suggestion
                mlog.log('  -- Skipping', mlog.bold(key), 'because it is too complex to modify')
```
Is there any chance `meson rewrite` can be made more powerful here? Currently, with (for example)
```meson
project('example',
  version : run_command('date', '+%s').stdout().strip())
```
it is impossible to do `meson rewrite kwargs set project '/' version '1.0.0'`, which makes the technique suggested in https://github.com/mesonbuild/meson/issues/688#issuecomment-1907635682 impossible.

Reply:
Yes, I think so. But I would do such a thing in a seperate PR if this one gets merged.

Reply:
Of course `meson rewrite` could be made more powerful, but I would do such a thing in a seperate PR if this one gets merged.

Reply:
Already awaiting review: #12782. In contrast to the current Merge request, that one is not a major rewrite, but a minor simplification of the code.